### PR TITLE
NAS-114605 / 22.12 / Added IxPermissionsComponent

### PR DESCRIPTION
**Pull request by RehanY147**
Go to System Settings -> General -> GUI Settings

Test the IxPermissions input in the form. Since the actual form submission probably won't succeed. In order to see changed values show on form start, you will have to update default value in code. Also test validation on the new component.
Also added testing.

Changes in this form must be removed before merging!!

### CODE REVIEW

File: src/app/modules/ix-forms/components/ix-permissions/ix-permissions.component.scss [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  }
+}
+
+.label-container {
+  margin-bottom: 1px;
+  margin-left: 1px;
+}
+
+.label {
+  font-size: x-small;
```
Comment (denysbutenko) [inline src/app/modules/ix-forms/components/ix-permissions/ix-permissions.component.scss:44]: it makes all labels smaller than others in a form:
image
image

Comment (RehanY147) [reply src/app/modules/ix-forms/components/ix-permissions/ix-permissions.component.scss:44]: It looks consistent with the other inputs on the new ix-forms.
image

Comment (AlexKarpov98) [reply src/app/modules/ix-forms/components/ix-permissions/ix-permissions.component.scss:44]: Screenshot_40

Looks label is small . I think it's better to make same font-size with above checkbox label `Show console Messages` considering new type of input.

File: src/app/pages/system/general-settings/gui-form/gui-form.component.ts [hunk 2]
```diff
@@ -52,6 +52,7 @@ export class GuiFormComponent {
     crash_reporting: [false, [Validators.required]],
     usage_collection: [false, [Validators.required]],
     ui_consolemsg: [false, [Validators.required]],
+    ui_perms: ['235', Validators.required],
```
Comment (AlexKarpov98) [inline src/app/pages/system/general-settings/gui-form/gui-form.component.ts:55]: Seems `Validators.required` does not work.

**Comment by RehanY147** [comment]
@AlexKarpov98 I'm not sure what `required` would mean for this input. At the very least, the input will give the value `'000'`. So, the input always has a value.

**Review by denysbutenko — APPROVED** [review]
Tested in User Form. Works well 👍🏻

**Review by AlexKarpov98 — APPROVED** [review]
LGTM! 👍

**Comment by denysbutenko** [comment]
https://github.com/truenas/webui/pull/6526 - you can test ix-permissions in this PR if you want

**Pull request merged**